src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -2361,6 +2371,53 @@ internal bool LoadModuleManifestData(ExternalScriptInfo scriptInfo, ManifestProc
                 if (bailOnFirstError) return null;
             }
 
+// Add PSEdition check for Windows PowerShell
+// modules imported from the System32 path
+#if !UNIX
```

Comment:
I think the `# if UNIX` part can be abstracted in `IsPSEditionCompatible`, and then code here don't need to be divided up.
The `IsPSEditionCompatible` would look like this:
```powershell
        private bool IsPSEditionCompatible(
            string moduleManifestPath,
            IEnumerable<string> compatiblePSEditions,
            out bool isOnSystem32ModulePath)
        {
            isOnSystem32ModulePath = false;
#if UNIX
            return false;
#else
            string windowsPSModulePath = ModuleIntrinsics.GetWindowsPowerShellPSHomeModulePath();
            if (!moduleManifestPath.StartsWith(windowsPSModulePath, StringComparison.OrdinalIgnoreCase))
            {
                return true;
            }

            isOnSystem32ModulePath = true;

            return BaseSkipEditionCheck || Utils.IsPSEditionSupported(compatiblePSEditions);
#endif
        }
```

Reply:
Yes I thought I'd start with the "zero runtime overhead on UNIX" approach and see how everyone felt. I definitely think your way is cleaner!